app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsPresenter.kt

```diff
@@ -0,0 +1,68 @@
+package ar.com.wolox.android.example.ui.example.home
+
+import ar.com.wolox.android.example.model.News
+import ar.com.wolox.android.example.network.NewsService
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import ar.com.wolox.wolmo.networking.retrofit.callback.NetworkCallback
+import okhttp3.ResponseBody
+import java.util.LinkedList
+import javax.inject.Inject
+
+class HomeNewsPresenter @Inject constructor() : BasePresenter<IHomeNewsView>() {
+
+    @Inject lateinit var retrofitService: RetrofitServices
+
+    fun onInit() {
+        retrofitService.getService(NewsService::class.java).getNews().enqueue(
+                object : NetworkCallback<List<News>>() {
+
+                    override fun onResponseSuccessful(news: List<News>?) {
+                        view.onNewsGotFromDatabase(triplicateNews(news))
+                    }
+
+                    override fun onCallFailure(throwable: Throwable) {
+                        // TODO: if is necessary, we can call a function on view to make a Toast
+                    }
+
+                    override fun onResponseFailed(body: ResponseBody?, code: Int) {
+                        // TODO: if is necessary, we can call a function on view to make a Toast
+                    }
+                }
+        )
+    }
+
+    fun onRefresh() {
```

Comment:
Well, I don't understand why there are two requests to bring news

Reply:
Asuming that the API had pagination implemented, the original idea was to have a function to get the next page of pagination (every next page will have older news than the last page gotten) and other to get recent news on refresh, this will be newer news, so is equivalent to ask for newer pages. The function **onInit()** must be called **getNextPage(Int)**

But the API does not support pagination, so it sound ridiculous now :sweat_smile: